core/contextmenu_registry.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Registry for context menu option items.
+ * @author maribethb@google.com (Maribeth Bottorff)
+ */
+'use strict';
+
+/**
+ * @name Blockly.ContextMenuRegistry
+ * @namespace
+ */
+goog.provide('Blockly.ContextMenuRegistry');
+
+/**
+ * Where this menu item should be rendered. If the menu item should be rendered in multiple
+ * scopes, e.g. on both a block and a workspace, it should be registered for each scope.
+ * @enum {string}
+ */
+Blockly.ContextMenuRegistry.ScopeType = {
+  BLOCK: 'block',
+  WORKSPACE: 'workspace',
+};
+
+/**
+ * The actual workspace/block where the menu is being rendered. This is passed to callback and
+ * displayText functions that depend on this information.
+ * @typedef {{
+ *    block: (Blockly.BlockSvg|undefined),
+ *    workspace: (Blockly.WorkspaceSvg|undefined),
+ * }}
+ */
+Blockly.ContextMenuRegistry.Scope;
+
+/**
+ * A menu item as entered in the registry.
+ * @typedef {{
+ *    callback: function(*),
```

Comment:
Is the type of the callback parameter `Blockly.ContextMenuRegistry.ScopeType` as well, similar to other methods here?

Reply:
Fixed